audiobook-reading-order-ids/README.md

```diff
@@ -0,0 +1,257 @@
+
+Reading order identifiers must have the following properties:
+
+* Identifiers must be _unique_ within a given book. No two reading order items can have the same
+  identifier.
+* Identifiers must be _stable_. Some manifest formats (notably Overdrive) contain information that
+  has a short expiration date and must be continuously refreshed during use. If identifiers were
+  to be derived from this short-lived content, then it would be impossible to place the identifiers
+  inside long-lived data such as bookmarks, as the identifiers would become useless soon after
+  creation.
+* The generation of identifiers must be _deterministic_. There must be a straightforward,
+  deterministic algorithm that can generate identifiers; for a given manifest `m`, the set
+  of identifiers generated for `m` must not vary between generations.
+
+## Definition
+
+A _reading order identifier_ should be considered to be an opaque string value in no particular
+format. Two identifiers `a` and `b` are equal iff all of the following conditions hold:
+
+* The length of `a` is equal to the length of `b`.
+* For identifiers `a` and `b` of length `n`, for each `i` in `[0, n)`, `a[i] = b[i]`.
+
+Identifiers have no defined order; no identifier should be considered to be "less than" or
+"greater than" any other identifier.
+
+Identifiers may occasionally look like [URI](https://datatracker.ietf.org/doc/html/rfc3986)
+values, but should not be treated as such: There are no guarantees that an identifier is unique
+across all possible manifests; identifiers are only guaranteed to be unique within a particular
+manifest.
+
+For reasons of sanity, identifiers must be of a non-zero length.
+
+## Generation
+
+The following sections describe the rules that should be used to generate identifiers for different
+types of manifests. Applications should try each rule in turn, falling back to the [Fallback](#fallback)
+rule if none of the preceding rules apply.
+
+### WebPub Manifests
+
+Manifests in [WebPub manifest](https://github.com/readium/webpub-manifest) format are required to
+contain a `readingOrder` array where each object in the array contains a `href` property that
+is expected to be unique within that `readingOrder` array.
+
+If the object does _not_ also contain a `templated` property, then the reading order identifier
+should be equal to the value of the `href` property. The reason for avoiding the use of `href`
+values that are marked as `templated` is that, in the presence of URI templating, the values of
+the `href` field may not actually be unique: It's permitted (although unusual and obscure) for
+every `templated` `href` value to be the same, and for each reading order item to contain a
+`properties` object containing values to be substituted into the `href` in order to produce a
+unique URI. This is a level of complexity that we currently do not want to mandate that players
+handle correctly, due to the apparent almost nonexistent use of `templated` links.
```

Comment:
Honestly given that we don't see templated links in any manifests, it seems like it would be fine to drop this section. 

From what I've seen of webpubs with templates in them, what is described in this section doesn't match how they are used. The properties object is not related to the template in the URI. The information about what can live in the properties is here: https://readium.org/webpub-manifest/properties.html

Whats in the template is left to the particular type of link, and up to the implementation, from what I've seen. Like in ODL they define a list of template parameters and what they mean [1]. And in OPDS2 they define a templated search link that has a query parameter [2], but both of these don't work quite like this section describes. 

[1] - https://drafts.opds.io/odl-1.0.html#52-uri-template-of-a-checkout-link
[2] - https://drafts.opds.io/opds-2.0#2-search

Given the uncertainty around templated links in audiobook manifests, I think ignoring them is fine, or defining them as an error case. We can devise an implementation if / when the need arises.

Reply:
I'd be happier making this an error than dropping this section entirely. If this section is dropped, and we encounter a manifest with templated URIs, we're more likely to silently do the wrong thing. I'm happy to drop the description of the `properties` object if this isn't how it works in practice.

I'm not sure _where_ we'd specify "if you see a templated URI in your `readingOrder`, set fire to it and push it off a cliff".

Reply:
> we're more likely to silently do the wrong thing

I'm not sure this is the case. It depends how the templates are used, but imagine this use of templated URIs:

- https://example.com/get_ab_file?identifier=123456&token={secret}
  - Where `secret` is some limited time secret that you template into the URI in order to retrieve the file for a short amount of time.  

In this case its perfectly reasonable to use `href` even though it contains template parameters. So it really depends on how the template gets used.

Reply:
In my cache of demarque manifests, there are not any with the `templated` property set, so its an extreme edge case. So really I'm good with handling this however we would like. Maybe we just leave it as a fallback case, as you've written it, but update the text describing the templated parameter to remove the mention of `properties`?

Reply:
I've made this into a hard error.